custom-metrics-stackdriver-adapter/pkg/adapter/provider/translator.go

```diff
@@ -84,6 +85,49 @@ func (t *Translator) GetRespForPod(response *stackdriver.ListTimeSeriesResponse,
 	return nil, fmt.Errorf("Illegal state")
 }
 
+// GetSDReqForPods returns Stackdriver request for query for multiple pods.
+func (t *Translator) GetSDReqForPods(podList runtime.Object, metricName string) (*stackdriver.ProjectsTimeSeriesListCall, error) {
+	resourceIDs, err := getResourceIDs(podList)
+	if err != nil {
+		return nil, err
+	}
+	if len(resourceIDs) == 0 {
+		return nil, fmt.Errorf("No objects matching provided selector")
+	}
+	project := fmt.Sprintf("projects/%s", t.config.Project)
+	endTime := t.clock.Now()
+	startTime := endTime.Add(-t.reqWindow)
+	filter, err := t.createFilter(resourceIDs, t.config.Project, t.config.Zone, t.config.Cluster, metricName)
+	if err != nil {
+		return nil, err
+	}
+	return t.service.Projects.TimeSeries.List(project).Filter(filter).
+		IntervalStartTime(startTime.Format(time.RFC3339)).
+		IntervalEndTime(endTime.Format(time.RFC3339)).
+		AggregationPerSeriesAligner("ALIGN_NEXT_OLDER").
+		AggregationAlignmentPeriod(fmt.Sprintf("%vs", int64(t.reqWindow.Seconds()))), nil
+}
+
+// GetRespForPods returns translates Stackdriver response to a Custom Metric associated
+// with multiple pods.
+func (t *Translator) GetRespForPods(response *stackdriver.ListTimeSeriesResponse, podList runtime.Object, groupResource schema.GroupResource, metricName string, namespace string) (*custom_metrics.MetricValueList, error) {
```

Comment:
same here

Reply:
1. Done
2. This is a little different than GetReq*, because the results are different, i.e. have different type. I think that all common logic is shared via functions getMetricValuesFromResponse() and metricFor().